packages/backend/src/services/UserService.ts

```diff
@@ -1150,4 +1154,64 @@ export class UserService {
             },
         });
     }
+
+    async getLoginOptions(email: string): Promise<LoginOptions> {
+        const openIdIssuer = await this.userModel.getOpenIdIssuer(email);
+        if (openIdIssuer !== undefined) {
+            const getRedirectUri = (issuer: OpenIdIdentityIssuerType) => {
+                switch (issuer) {
+                    case OpenIdIdentityIssuerType.AZUREAD:
+                        return this.lightdashConfig.auth.azuread.loginPath;
+                    case OpenIdIdentityIssuerType.GOOGLE:
+                        return this.lightdashConfig.auth.google.loginPath;
+                    case OpenIdIdentityIssuerType.OKTA:
+                        return this.lightdashConfig.auth.okta.loginPath;
+                    case OpenIdIdentityIssuerType.ONELOGIN:
+                        return this.lightdashConfig.auth.oneLogin.loginPath;
+                    default:
+                        assertUnreachable(
+                            issuer,
+                            `Invalid login option for issuer ${issuer}`,
+                        );
+                }
+                return undefined;
+            };
+
+            return {
+                showOptions: [openIdIssuer],
+                forceRedirect: true,
+                redirectUri: new URL(
+                    `/api/v1${getRedirectUri(
+                        openIdIssuer,
+                    )}?login_hint=${email}`,
```

Comment:
should we encodeURLComponent this?

Reply:
done 
![Screenshot from 2024-03-14 08-43-01](https://github.com/lightdash/lightdash/assets/1983672/7ae762f1-8f66-4be5-adee-61f58e7fd4c9)